syncstorage/src/web/extractors.rs

```diff
@@ -1187,13 +1185,35 @@ impl From<Offset> for params::Offset {
 }
 
 impl FromStr for Offset {
-    type Err = ParseIntError;
+    type Err = ApiError;
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         // issue559: Disable ':' support for now: simply parse as i64 as
         // previously (it was u64 previously but i64's close enough)
+        //
+        // We were seeing a number of errors where iOS clients would return
+        // offsets that contained ':' (e.g. '123456790:123') This is a valid
+        // form that the original python server used, but is not implemented
+        // with this version. Returning a 412 was suggested in order to
+        // break any loop that might exist on the client.
+        if s.contains(':') {
+            return Err(ValidationErrorKind::FromDetails(
```

Comment:
Hmm, is the error returned here being overwritten by the [`map_err` call on line 1288](https://github.com/mozilla-services/syncstorage-rs/blob/54321126d81710d2150f2b7d05b2a1ccb0338e89/syncstorage/src/web/extractors.rs#L1288)? If so, I think that would overwrite the 412 error code with a 400 (I made some changes to the `test_invalid_offset` unit test, and I think I was able to confirm this).

I think we may need to pass through the name and label from the original offset error to the one on line 1288?